doc/api/dns.md

```diff
@@ -73,14 +76,59 @@ The difference between `REVERSE_A`, `REVERSE_AAAA` and `PTR` is that
 an IPv6 address, while for `PTR` you need to construct yourself the
 reversed IP address name to query.
 
-The callback could be called immediately if an error occurs while
-sending the query to the server.
+In case of success, the error argument of the callback is passed an
+instance of `mk::NoError()`. Otherwise, the error that occurred is
+reported. Among all the possible errors, the following are defined by
+MeasurementKit DNS implementation:
+
+
+- `FormatError`: invalid response format
+- `ServerFailedError`:  server failure
+- `NotExistError`:  the name does not exist
+- `NotImplementedError`:  query not implemented
+- `RefusedError`:  the server refuses to reply
+- `TruncatedError`:  response truncated
+- `UknownError`:  internal evnds error
+- `TimeoutError`:  query timed out
+- `ShutdownError`:  evdns library was shut down
+- `CancelError`:  user cancelled query
+- `NoDataError`:  no data in the response
+
+
+The `Response` class holds the result of an async DNS `Query`. You do not
+typically instantiate a `Response` yourself, rather you receive an instance
+of `Response` in response to a DNS query.
+
+Given a DNS response data can be retrived as it follows:
```

Comment:
Ah, right. Well probably in that case it fits where it is because the structure is explicitly listed (a thing that we cannot move in the synopsis) while here the focus is on usage (since the synopsis shows usage, I was proposing to move that usage-related part into the synopsis).